### ISSUE TITLE
Speed up on dnn-based lr

### ISSUE THREAD

**Issue by yankang18** [FederatedAI/FATE#68]
Currently, too much running time was spent on decryption, which is account for 66% of the running time per iteration. 

The reason that causes this issue is that currently, we delegate the whole work of updating local neural network model (calculate gradients and update model parameters) to tensorflow for the purpose of saving engineering effort and supporting model extensibility. To let tensorflow automatically update local model, we need to feed tensorflow with plain gradients of all sample. That is, if we have 30,000 samples, we would have 30,000 samples of gradients. The bottleneck comes when we decrypt the whole 30,000 samples of gradients (this is not safe in terms of data protection), which consumes lots of time on both computation and communication.

The solution to solve this issue is : 
1)	Use eggroll to parallelly decrypt gradients 
2)	Compute the new gradients used to update local model manually（do not use tensorflow for this part of work and tensorflow does not support this for now）.
